modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/FilterQueryRequestProcessor.java

```diff
@@ -0,0 +1,117 @@
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchRequestProcessor;
+
+import java.io.InputStream;
+import java.util.Map;
+
+import static org.opensearch.index.query.AbstractQueryBuilder.parseInnerQueryBuilder;
+
+/**
+ * This is a {@link SearchRequestProcessor} that replaces the incoming query with a BooleanQuery
+ * that MUST match the incoming query with a FILTER clause based on the configured query.
+ */
+public class FilterQueryRequestProcessor extends AbstractProcessor implements SearchRequestProcessor {
+    /**
+     * Key to reference this processor type from a search pipeline.
+     */
+    public static final String TYPE = "filter_query";
+
+    final QueryBuilder filterQuery;
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    /**
+     * Constructor that takes a filter query.
+     *
+     * @param tag         processor tag
+     * @param description processor description
+     * @param filterQuery the query that will be added as a filter to incoming queries
+     */
+    public FilterQueryRequestProcessor(String tag, String description, QueryBuilder filterQuery) {
+        super(tag, description);
+        this.filterQuery = filterQuery;
+    }
+
+    @Override
+    public SearchRequest processRequest(SearchRequest request) throws Exception {
+        QueryBuilder originalQuery = null;
+        if (request.source() != null) {
+            originalQuery = request.source().query();
+        }
+
+        BoolQueryBuilder filteredQuery = new BoolQueryBuilder().filter(filterQuery);
+        if (originalQuery != null) {
+            filteredQuery.must(originalQuery);
+        }
+        if (request.source() == null) {
+            request.source(new SearchSourceBuilder());
+        }
+        request.source().query(filteredQuery);
+        return request;
+    }
+
+    static class Factory implements Processor.Factory {
+        private final NamedXContentRegistry namedXContentRegistry;
+        public static final ParseField QUERY_FIELD = new ParseField("query");
+
+        Factory(NamedXContentRegistry namedXContentRegistry) {
+            this.namedXContentRegistry = namedXContentRegistry;
+        }
+
+        @Override
+        public FilterQueryRequestProcessor create(
+            Map<String, Processor.Factory> processorFactories,
```

Comment:
This is needed when we start doing composition of processors (i.e. a processor with a nested processor).

Since it's part of the "public" interface for plugin developers, this is a case where I want to aim for where we plan to go, rather than where we are today.